what-changed/what_changed_test.go

```diff
@@ -42,6 +42,125 @@ func TestCompareSwaggerDocuments(t *testing.T) {
 	assert.Equal(t, 27, changes.TotalBreakingChanges())
 }
 
+// TestCacheCollisionSelfReference reproduces the cache collision bug with self-referencing schemas
+// This is the key pattern that triggers the QuickHash cache collision issue
+func TestCacheCollisionSelfReference(t *testing.T) {
+	// Original spec - TreeNode schema with basic properties
+	original := `{
+  "openapi": "3.0.3",
+  "info": {"title": "Test API", "version": "1.0.0"},
+  "paths": {
+    "/tree": {
+      "get": {
+        "responses": {
+          "200": {
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/TreeNode"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "TreeNode": {
+        "type": "object",
+        "additionalProperties": false,
+        "properties": {
+          "children": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/TreeNode"
```

Comment:
The self reference here causes the diff to always report no change (despite there being obvious changes).